examples/chip-tool/commands/pairing/PairingCommand.cpp

```diff
@@ -404,8 +416,26 @@ void PairingCommand::OnICDRegistrationComplete(NodeId nodeId, uint32_t icdCounte
     chip::Encoding::BytesToHex(mICDSymmetricKey.Value().data(), mICDSymmetricKey.Value().size(), icdSymmetricKeyHex,
                                sizeof(icdSymmetricKeyHex), chip::Encoding::HexFlags::kNullTerminate);
 
-    // TODO: Persist symmetric key.
+    chip::app::ICDClientInfo clientInfo;
+    clientInfo.peer_node         = chip::ScopedNodeId(nodeId, CurrentCommissioner().GetFabricIndex());
+    clientInfo.monitored_subject = mICDMonitoredSubject.Value();
+
+    CHIP_ERROR err = GetICDClientStorage().SetKey(clientInfo, mICDSymmetricKey.Value());
+    if (err == CHIP_NO_ERROR)
+    {
+        err = GetICDClientStorage().StoreEntry(clientInfo);
+    }
+
+    if (err != CHIP_NO_ERROR)
+    {
+        // StoreEntry is unlikely to fail since it is a local operation for CHIP Tool.
```

Comment:
Why don't we stop assuming that and actually write this code correctly?  If StoreEntry fails, we should remove the key, etc....

Reply:
yes, agree, updated, thanks

Reply:
If StoreEntry fails, why should we need to remove the key (from client)?

Maybe we can just delete this comment. Since we are not at the final stage of commissioning, quit here and the device should wipe its record after failsafe timed out. (Since this is commissioning)

Reply:
Ok, `SetKey` actually persists the key, so we should delete the key if StoreEntry() failed.